### ISSUE TITLE
PolyLine bug at longitude 180°

### ISSUE THREAD

**Issue by stellardeveloper** [MKergall/osmbonuspack#167]
Hi,

im am writing apps to display large global lines (eg. intercontinental flight tracks, solar eclipse boundaries and similar)
There is a problem with geodetic=true Polylines crossing the +-180° longitude border:
Only the small segment crossing the limit will be drawn, the rest of the line is invisible.
Example:

```
    GeoPoint startPoint =   new GeoPoint(50.0, 160.0);
    GeoPoint endPoint   =   new GeoPoint(65.0, 190.0);
    ArrayList<GeoPoint> waypoints = new ArrayList<GeoPoint>();
    waypoints.add(startPoint);
    waypoints.add(endPoint);

    Polyline line = new Polyline (this);
    line.setGeodesic(true);
    line.setPoints(waypoints);
```

If the line is far away from the 180° limit, there are no problems, for example with:

```
    GeoPoint startPoint =   new GeoPoint(50.0, 60.0);
    GeoPoint endPoint   =   new GeoPoint(65.0, 90.0);
```

There is another problem with Polylines at high latitudes near the limit, disappearing at low zoom leveles:

```
    GeoPoint startPoint =   new GeoPoint(50.0, 60.0);
    GeoPoint endPoint   =   new GeoPoint(65.0, 170.0);
```

This will be visible only at zoom levels >= 4. This problem seems independent of setGeodesic(true) or false and it affects the old deprecated PathOverlay as well. The same line near the prime meridian will only disappear at zoom level 1 and below. 

Im a using:
Android 4.0.4, developing with eclipse
osmdroid-android-4.3.jar
osmbonuspack_v5.1.jar

BTW: is there a way to provide JAR files of the osmbonuspack > 5.1? 
Extracting the classes.jar from the bonuspack.aar does not work.

Kind regards

**Comment by MKergall** [comment]
3 issues. 

1) The +-180° longitude issue 
=> Effectively, it's not handled (probably other issues around this subject at other places). Are you ready to implement the fix?... I would include it in the lib. 

2) At high latitudes near the limit, Polyline disappearing at low zoom level 
=> could be due to lack of precision on GeoPoint. I will check. 

3) Is there a way to provide JAR files of the osmbonuspack > 5.1 instead of AAR ? 
=> If really needed: compile the lib yourself on Eclipse, as a jar. There will be one or two issues to fix around the default bubble. 
=> Or (recommended) move to Android Studio. Inescapable: osmdroid 5.0.1 is now delivered as an AAR as well.

**Comment by spyhunter99** [comment]
osmdroid switched to AAR because it an android specific library and to
include the default icon sets, which just caused a lot of confusion. it
simply can't run in the JDK as such, AAR makes more sense.

2nd the android studio bit. The google overlords have been gracious enough
to choose our IDE and build system for us. Down with choice and options and
flexibility and preference and speed and....

**Comment by spyhunter99** [comment]
In addition, I've see this issue too, only sometimes lines near the cross
over points will sometimes unexpectedly been drawn across the entire globe.
It looks strange to say the least, but it usually self corrects with
zooming and panning

**Comment by MKergall** [comment]
I confirm issue 2). 
Even with classical latitudes (Greece) and small lines (10 km). 
I create a separate issue #168.

**Comment by MKergall** [comment]
And there is another issue related to Issue 2) => I create Issue #169

**Comment by MKergall** [comment]
In fact, Issue 1) was probably linked to issue #169, whose fix is now committed. 

@stellardeveloper : Using this fix, I tried extensively your polyline (50.0, 160.0 - 65.0, 190.0), with or witout setGeodesic. 
Could you try again using the fix?

BTW, (65.0, 190.0) is not standard. You should set (65.0, -170.0)

**Comment by stellardeveloper** [comment]
Hello,
the primary problem is solved on my device, the polyline does not disappear any longer. However, on very low zoom levels (level 1 and below), it long lines have to extend the image borders and have to reappear on the other side, the line will be broken and connected. Sometimes there will appear weird line paterns changing funny on moving the view, see screenshot attached.
![screenshot_2015-11-23-19-19-21](https://cloud.githubusercontent.com/assets/15923908/11345587/595cd402-9218-11e5-9229-e619f188fc8d.png)
Image above: 
Green line: from GeoPoint(50.0, 60.0)  to GeoPoint(65.0, 150.0), geodetic,
Yellow line: from  GeoPoint(65.0, 150.0) to GeoPoint(-65.0, -100.0), geodetic,
blue line: same as yellow, non-geodetic.

BTW: as written in PM, I got a resource problem with the resource proxy. I solved it by replacing marker.java line 87, instead of using "resourceProxy.getDrawable" use: 
    mDefaultIcon = mapView.getResources().getDrawable( R.drawable.marker_default );
=> open new issue?
